src/wallet/wallet.cpp

```diff
@@ -3851,6 +3851,9 @@ bool CWallet::Verify(std::string wallet_file, bool salvage_wallet, std::string&
         }
     }
 
+    // Keep same database environment instance across Verify/Recover calls below.
+    std::unique_ptr<WalletDatabase> database = WalletDatabase::Create(wallet_path);
```

Comment:
This declaration hides the member `CWallet::database`. Is that intentional? If not, use another name? :-)